service/src/main/resources/templates/specimen_compound_statement_template.mustache

```diff
@@ -10,13 +10,32 @@
         <specimen typeCode="SPC">
             <specimenRole classCode="SPEC">
                 <id root="37252CF4-D7F6-4CBA-89B2-63477F0C8374"/>
-                <id root="2.16.840.1.113883.2.1.4.5.2" extension="{{accessionIdentifier}}"/>
-                {{{effectiveTime}}}
+                {{#accessionIdentifier}}
+                <id root="2.16.840.1.113883.2.1.4.5.2" extension="{{{.}}}"/>
+                {{/accessionIdentifier}}
+                <effectiveTime>
```

Comment:
There's a note on the spreadsheet saying this should not be generated if there neither collection.collected[x] or receivedTime. So maybe we need `{{#effectiveTime}}` around the `<effectiveTime>` tag here?

Reply:
in line 118 there is note that it should be nullFlavor in case the date is not provided. @dmcavenue1 do you know what approach is the final one?

I see I am missing test for that.

Reply:
yeah I saw the note about adding `nullFlavor` too

Reply:
@n-r-b you are right, I updated the mapping and added test